src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php

```diff
@@ -503,6 +503,23 @@ protected function castAttribute($key, $value)
         }
     }
 
+    /**
+     * Cast a null attribute.
+     *
+     * @param  string  $key
+     * @param  mixed  $value
+     * @return mixed
+     */
+    protected function castNullAttribute($key, $value)
+    {
+        switch ($this->getCastType($key)) {
```

Comment:
Why not a simple `if... then`?

Reply:
Just for extendability (following castAttribute() function's pattern)

Reply:
As you can found the proposal on the issue

Following types might be allowed in the future (I guess)
If you think this can cause performance issue, then I'll change it to simple `if... then`
```
/**
 * Cast a null attribute.
 *
 * @param  string  $key
 * @param  mixed  $value
 * @return mixed
 */
protected function castNullAttribute($key, $value)
{
    switch ($this->getCastType($key)) {
        case 'array':
            return [];
        case 'collection':
            return new BaseCollection;
        default:
            return $value;
    }
}
```